service/tablet_allocator.cc

```diff
@@ -925,12 +922,12 @@ class load_balancer {
 
                     const locator::node& target_node = topo.get_node(new_target);
                     const locator::node& source_node = topo.get_node(src_host);
-                    if (target_node.dc_rack().rack != source_node.dc_rack().rack
-                            && (rack_load[target_node.dc_rack().rack] + 1 > max_rack_load)) {
+                    if (target_node.rack() != source_node.rack()
+                            && (rack_load[target_node.rack()] + 1 > max_rack_load)) {
```

Comment:
@tgrabiec how did this work previously for nodes in different data centers that happen to use the same rack name?

Reply:
target_node and source_node are always in the same datacenter, we don't migrate across DCs.

Reply:
I see.  Well, I wouldn't count on that here.
The new code is resilient to that.